library/core/src/ptr/const_ptr.rs

```diff
@@ -66,6 +66,32 @@ impl<T: ?Sized> *const T {
         self as _
     }
 
+    /// Try to cast to a pointer of another type by checking aligment.
+    ///
+    /// If the pointer is well aligned to the target type, it will be casted
+    /// to the target type. Otherwise, `None` is returned.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// #![feature(pointer_try_cast_aligned)]
+    ///
+    /// // On some platforms, the alignment of i32 is less than 4.
+    /// #[repr(align(4))]
+    /// struct AlignedI32(i32);
+    ///
+    /// // This pointer is aligned to 4 bytes
+    /// let ptr: *const u8 = 0x1000 as _;
+    /// assert!(ptr.try_cast_aligned::<AlignedI32>().is_some());
```

Comment:
I think you can drop the `AlignedI32` to keep the example cleaner:

```rust
    /// let ptr = 0x1000 as *const u8;
    ///
    /// // i32 has at most 4-byte alignment, so this will succeed
    /// assert!(ptr.try_cast_aligned::<i32>().is_some());
```

It would also be good to demonstrate a case where `None` is returned, e.g. the same thing with 0x1001.